src/main/groovy/com/avast/gradle/dockercompose/tasks/ComposeUp.groovy

```diff
@@ -48,11 +52,28 @@ class ComposeUp extends DefaultTask {
         }
     }
 
+    protected void initializeComposeFilesInfo() {
+        composeFiles = extension.useComposeFiles
+        filesSpecified = composeFiles && composeFiles.size() > 0
+    }
+
+    protected void honourComposeFileOverride() {

```

Comment:
See my earlier comment on `composeFiles`.
The `filesSpecified` is a shorthand because it is used multiple times.

Also `docker-compose up -d` behaves differently than `docker-compose -f docker-compose.yml up -d`. In the latter case the docker-compose.override.yml is ignored

I moved the code in `honourComposeFileOverride` to `loadServicesInfo` (though can also be in `getServiceNames`.